include/deal.II/numerics/vector_tools_evaluate.h

```diff
@@ -343,6 +343,42 @@ namespace VectorTools
 
 
 
+    /**
+     * Perform reduction for tensors of tensors (e.g., gradient of
+     * vectorial quantities).
+     */
+    template <int n_components, int rank, int dim, typename Number>
+    Tensor<1, n_components, Tensor<rank, dim, Number>>
+    reduce(
+      const EvaluationFlags::EvaluationFlags &flags,
+      const ArrayView<const Tensor<1, n_components, Tensor<rank, dim, Number>>>
+        &values)
+    {
+      switch (flags)
+        {
+          case EvaluationFlags::avg:
+            {
+              Tensor<1, n_components, Tensor<rank, dim, Number>> temp;
+
+              for (unsigned int j = 0; j < values.size(); ++j)
+                for (unsigned int i = 0; i < n_components; ++i)
+                  temp[i] = temp[i] + values[j][i];
+
+              for (unsigned int i = 0; i < n_components; ++i)
+                temp[i] /= (Number(1.0) * values.size());
```

Comment:
Does it work to simply write
```suggestion
                temp[i] /= Number(values.size());
```